// @patternfly/react-core/dist/umd/components/OptionsMenu/OptionsMenu.js
(function (global, factory) {
  if (typeof define === "function" && define.amd) {
    define(["exports", "prop-types", "react", "@patternfly/react-styles/css/components/OptionsMenu/options-menu", "@patternfly/react-styles"], factory);
  } else if (typeof exports !== "undefined") {
    factory(exports, require("prop-types"), require("react"), require("@patternfly/react-styles/css/components/OptionsMenu/options-menu"), require("@patternfly/react-styles"));
  } else {
    var mod = {
      exports: {}
    };
    factory(mod.exports, global.propTypes, global.react, global.optionsMenu, global.reactStyles);
    global.undefined = mod.exports;
  }
})(this, function (exports, _propTypes, _react, _optionsMenu, _reactStyles) {
  "use strict";

  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.OptionsMenu = exports.OptionsMenuDirection = exports.OptionsMenuPosition = undefined;

  var _propTypes2 = _interopRequireDefault(_propTypes);

  var React = _interopRequireWildcard(_react);

  var _optionsMenu2 = _interopRequireDefault(_optionsMenu);

  function _interopRequireWildcard(obj) {
    if (obj && obj.__esModule) {
      return obj;
    } else {
      var newObj = {};

      if (obj != null) {
        for (var key in obj) {
          if (Object.prototype.hasOwnProperty.call(obj, key)) {
            var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {};

            if (desc.get || desc.set) {
              Object.defineProperty(newObj, key, desc);
            } else {
              newObj[key] = obj[key];
            }
          }
        }
      }

      newObj.default = obj;
      return newObj;
    }
  }

  function _interopRequireDefault(obj) {
    return obj && obj.__esModule ? obj : {
      default: obj
    };
  }

  function _extends() {
    _extends = Object.assign || function (target) {
      for (var i = 1; i < arguments.length; i++) {
        var source = arguments[i];

        for (var key in source) {
          if (Object.prototype.hasOwnProperty.call(source, key)) {
            target[key] = source[key];
          }
        }
      }

      return target;
    };

    return _extends.apply(this, arguments);
  }

  function _objectWithoutProperties(source, excluded) {
    if (source == null) return {};

    var target = _objectWithoutPropertiesLoose(source, excluded);

    var key, i;

    if (Object.getOwnPropertySymbols) {
      var sourceSymbolKeys = Object.getOwnPropertySymbols(source);

      for (i = 0; i < sourceSymbolKeys.length; i++) {
        key = sourceSymbolKeys[i];
        if (excluded.indexOf(key) >= 0) continue;
        if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
        target[key] = source[key];
      }
    }

    return target;
  }

  function _objectWithoutPropertiesLoose(source, excluded) {
    if (source == null) return {};
    var target = {};
    var sourceKeys = Object.keys(source);
    var key, i;

    for (i = 0; i < sourceKeys.length; i++) {
      key = sourceKeys[i];
      if (excluded.indexOf(key) >= 0) continue;
      target[key] = source[key];
    }

    return target;
  }

  let OptionsMenuPosition = exports.OptionsMenuPosition = undefined;

  (function (OptionsMenuPosition) {
    OptionsMenuPosition["right"] = "right";
    OptionsMenuPosition["left"] = "left";
  })(OptionsMenuPosition || (exports.OptionsMenuPosition = OptionsMenuPosition = {}));

  let OptionsMenuDirection = exports.OptionsMenuDirection = undefined;

  (function (OptionsMenuDirection) {
    OptionsMenuDirection["up"] = "up";
    OptionsMenuDirection["down"] = "down";
  })(OptionsMenuDirection || (exports.OptionsMenuDirection = OptionsMenuDirection = {}));

  const OptionsMenu = exports.OptionsMenu = _ref => {
    let {
      className = '',
      isOpen = false,
      isPlain = false,
      ariaLabelMenu = '',
      direction = OptionsMenuDirection.down,
      position = OptionsMenuPosition.left,
      id,
      menuItems,
      toggle
    } = _ref,
        props = _objectWithoutProperties(_ref, ["className", "isOpen", "isPlain", "ariaLabelMenu", "direction", "position", "id", "menuItems", "toggle"]);

    return React.createElement("div", _extends({
      id: id,
      className: (0, _reactStyles.css)(_optionsMenu2.default.optionsMenu, direction === OptionsMenuDirection.up && (0, _reactStyles.getModifier)(_optionsMenu2.default, 'top'), position === OptionsMenuPosition.right && (0, _reactStyles.getModifier)(_optionsMenu2.default, 'align-right'), isOpen && (0, _reactStyles.getModifier)(_optionsMenu2.default, 'expanded'), className)
    }, props), React.Children.map(toggle, oneToggle => React.cloneElement(oneToggle, {
      parentId: id,
      isOpen,
      isPlain
    })), isOpen && React.createElement("ul", _extends({
      className: (0, _reactStyles.css)(_optionsMenu2.default.optionsMenuMenu, position === OptionsMenuPosition.right && (0, _reactStyles.getModifier)(_optionsMenu2.default, 'align-right'))
    }, ariaLabelMenu ? {
      'aria-label': ariaLabelMenu
    } : {
      'aria-labelledby': `${id}-toggle`
    }), menuItems));
  };

  OptionsMenu.propTypes = {
    className: _propTypes2.default.string,
    id: _propTypes2.default.string.isRequired,
    menuItems: _propTypes2.default.arrayOf(_propTypes2.default.node).isRequired,
    toggle: _propTypes2.default.element.isRequired,
    isPlain: _propTypes2.default.bool,
    isOpen: _propTypes2.default.bool,
    ariaLabelMenu: _propTypes2.default.string,
    position: _propTypes2.default.oneOf(['right', 'left']),
    direction: _propTypes2.default.oneOf(['up', 'down'])
  };
});
