// @patternfly/react-core/dist/js/components/Select/selectConstants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.KeyTypes = exports.SelectVariant = exports.SelectConsumer = exports.SelectProvider = exports.SelectContext = void 0;

var React = _interopRequireWildcard(require("react"));

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj["default"] = obj; return newObj; } }

var SelectContext = React.createContext(null);
exports.SelectContext = SelectContext;
var SelectProvider = SelectContext.Provider;
exports.SelectProvider = SelectProvider;
var SelectConsumer = SelectContext.Consumer;
exports.SelectConsumer = SelectConsumer;
var SelectVariant;
exports.SelectVariant = SelectVariant;

(function (SelectVariant) {
  SelectVariant["single"] = "single";
  SelectVariant["checkbox"] = "checkbox";
  SelectVariant["typeahead"] = "typeahead";
  SelectVariant["typeaheadMulti"] = "typeaheadmulti";
})(SelectVariant || (exports.SelectVariant = SelectVariant = {}));

var KeyTypes = {
  Tab: 'Tab',
  Space: ' ',
  Escape: 'Escape',
  Enter: 'Enter',
  ArrowUp: 'ArrowUp',
  ArrowDown: 'ArrowDown'
};
exports.KeyTypes = KeyTypes;
