// @patternfly/react-core/dist/js/components/Dropdown/dropdownConstants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DropdownArrowContext = exports.DropdownContext = exports.DropdownDirection = exports.DropdownPosition = void 0;

var React = _interopRequireWildcard(require("react"));

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj["default"] = obj; return newObj; } }

var DropdownPosition;
exports.DropdownPosition = DropdownPosition;

(function (DropdownPosition) {
  DropdownPosition["right"] = "right";
  DropdownPosition["left"] = "left";
})(DropdownPosition || (exports.DropdownPosition = DropdownPosition = {}));

;
var DropdownDirection;
exports.DropdownDirection = DropdownDirection;

(function (DropdownDirection) {
  DropdownDirection["up"] = "up";
  DropdownDirection["down"] = "down";
})(DropdownDirection || (exports.DropdownDirection = DropdownDirection = {}));

;
var DropdownContext = React.createContext({
  onSelect: function onSelect(event) {
    return undefined;
  },
  menuClass: '',
  itemClass: '',
  toggleClass: '',
  baseClass: '',
  baseComponent: 'div',
  sectionClass: '',
  sectionTitleClass: '',
  sectionComponent: 'section',
  disabledClass: '',
  hoverClass: '',
  separatorClass: ''
});
exports.DropdownContext = DropdownContext;
var DropdownArrowContext = React.createContext({
  keyHandler: null,
  sendRef: null
});
exports.DropdownArrowContext = DropdownArrowContext;
